src/App.jsx

```diff
@@ -14,33 +17,127 @@ export const goodsFromServer = [
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_ALPHABET = 'Sort alphabetically';
+const SORT_LENGTH = 'Sort by length';
+const REVERSE = 'Reverse';
+const RESET = 'Reset';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function styleClass(btn) {
+  switch (btn) {
+    case SORT_ALPHABET:
+      return 'is-info';
+    case SORT_LENGTH:
+      return 'is-success';
+    case REVERSE:
+      return 'is-warning';
+    case RESET:
+      return 'is-danger';
+    default:
+      return '';
+  }
+}
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+function sortGoods(goods, { typeField, isReversed }) {
+  const changedGoods = [...goods];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (typeField) {
+    if (typeField === SORT_ALPHABET) {
+      if (isReversed) {
+        return changedGoods.sort((a, b) => a.localeCompare(b)).reverse();
+      }
+
+      return changedGoods.sort((a, b) => a.localeCompare(b));
+    }
+
+    if (typeField === SORT_LENGTH) {
+      if (isReversed) return changedGoods.sort((a, b) => b.length - a.length);
+
+      return changedGoods.sort((a, b) => b.length - a.length).reverse();
+    }
+  }
+
+  if (!typeField && isReversed) {
+    changedGoods.reverse();
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return changedGoods;
+}
+
+export const App = () => {
+  const [sort, setSort] = useState({ typeField: '', isReversed: false });
+  const visibleGoods = sortGoods(goodsFromServer, sort);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() =>
+            setSort(prev => ({ ...prev, typeField: SORT_ALPHABET }))
+          }
+          type="button"
+          className={clsx(
+            'button',
+            styleClass(SORT_ALPHABET),
+            sort.typeField !== SORT_ALPHABET && 'is-light',
+          )}
+        >
+          Sort alphabetically
+        </button>
+        <button
+          onClick={() => {
+            setSort(prev => ({ ...prev, typeField: SORT_LENGTH }));
+          }}
+          type="button"
+          className={clsx(
+            'button',
+            styleClass(SORT_LENGTH),
+            sort.typeField !== SORT_LENGTH && 'is-light',
+          )}
+        >
+          Sort by length
+        </button>
+        <button
+          onClick={() => {
+            setSort(prev => {
+              const reverse = sort.isReversed !== true;
```

Comment:
- It doesn't make sense to compare with a boolean value as it's already boolean
- Get info not from sort, but from the previous value 
- Boolean variables should start from `is` or `has`